daml-assistant/src/DAML/Assistant/Install/Path.hs

```diff
@@ -22,19 +23,24 @@ import Graphics.Win32.Message
 import Graphics.Win32.Window.PostMessage
 import System.Win32.Registry hiding (regQueryValueEx)
 import System.Win32.Types
-#else
-import System.FilePath
 #endif
 
-updatePath :: (String -> IO ()) -> FilePath -> IO ()
+import DAML.Assistant.Types
+
+updatePath :: InstallOptions -> (String -> IO ()) -> FilePath -> IO ()
 #ifdef mingw32_HOST_OS
-updatePath output targetPath = do
+updatePath installOpts output targetPath
+    | SetPath b <- iSetPath installOpts, not b = suggestAddToPath output targetPath
+    | otherwise = do
     -- Updating PATH on Windows is annoying so we do it automatically.
     bracket (regOpenKeyEx hKEY_CURRENT_USER "Environment" kEY_ALL_ACCESS) regCloseKey $ \envKey -> do
-        path <- regQueryStringValue envKey "Path"
-        let paths = split (== ';') path
-        when (targetPath `notElem` paths) $ do
-            let newPath = intercalate ";" $ targetPath : paths
+        -- We use getSearchPath for the check instead of reading from the registry
+        -- since that allows users to modify the environment variable to temporarily
+        -- add it to PATH.
+        searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
+        when (targetPath `notElem` searchPaths) $ do
```

Comment:
You're testing the search path from `getSearchPath`, but then you are updating the registry value instead. Is `getSearchPath` always guaranteed to be a superset of the registry value? Otherwise you may end up updating the registry key multiple times. Would be better to just test the registry value, I think?

Reply:
That’s what I did before but it leads to modifying the registry even if the user either modified PATH directly, or added it to the global registry, … which sems like a bad idea.  I’ve now changed it to check both the registry and what we get from `getSearchPath` before modifying the registry which seems like a reasonably sane choice.

Reply:
Perfect, thanks! :+1: